ocaml/xapi/static_vdis.ml

```diff
@@ -77,7 +77,7 @@ let permanent_vdi_deactivate_by_uuid ~__context ~uuid =
   try
     let vdi = Db.VDI.get_by_uuid ~__context ~uuid in
     Sm.call_sm_vdi_functions ~__context ~vdi (fun srconf srtype sr ->
-        Sm.vdi_deactivate srconf srtype sr vdi
+        Sm.vdi_deactivate ~dbg:"" srconf srtype sr vdi
```

Comment:
instead of exposing and passing an empty string, perhaps we could use stronger typing by:
- making ~dbg an optional ?dbg in Debuginfo.with_dbg.
- inside Debuginfo.with_dbg, if ?dbg matches None, then use whatever is equivalent to "" for dbg, else use Some dbg.
- make ~dbg optional ?dbg in Sm.vdi_deactivate only, the other functions in Sm.* still use ~dbg.
- so this line in static_vdis.ml would work without passing dbg to Sm.vdi_deactivate.

Reply:
This makes `dbg` optional inside `Debuginfo.with_dgb` https://github.com/xapi-project/xen-api/pull/5293/commits/47170c82a588e218038284ce036dfdad2b1a597a. And this updates `Sm.vdi_deactivate` https://github.com/xapi-project/xen-api/pull/5293/commits/f2448fa4c67403a74d4f69ea259b8db6ddb5096a